modules/apps/transfer/module.go

```diff
@@ -254,14 +281,35 @@ func (am AppModule) OnChanOpenTry(
 	version,
 	counterpartyVersion string,
 ) error {
-	if err := ValidateTransferChannelParams(ctx, am.keeper, order, portID, channelID, version); err != nil {
-		return err
+	transferVersion, appVersion := channeltypes.SplitChannelVersion(version)
+	if transferVersion == "" {
+		// This may occur if:
+		// - transfer is acting as the base application in the middleware stack
+		// - the connected base application has chosen not to append its own
+		transferVersion = appVersion
 	}
 
-	if counterpartyVersion != types.Version {
-		return sdkerrors.Wrapf(types.ErrInvalidVersion, "invalid counterparty version: got: %s, expected %s", counterpartyVersion, types.Version)
+	cpTransferVersion, cpAppVersion := channeltypes.SplitChannelVersion(counterpartyVersion)
+	if cpTransferVersion == "" {
+		// This may occur if:
+		// - transfer is acting as the base application in the middleware stack
+		// - the connected base application has chosen not to append its own
+		cpTransferVersion = cpAppVersion
+	}
+
+	if err := ValidateTransferChannelParams(ctx, am.keeper, order, portID, channelID, transferVersion); err != nil {
+		return err
+	}
+	if cpTransferVersion != transferVersion {
+		return sdkerrors.Wrapf(types.ErrInvalidVersion, "expected counterparty version: %s, got: %s", types.Version, cpTransferVersion)
 	}
```

Comment:
I'd move this above the `ValidateTransferChannelParams` since it doesn't require stateful validation